pkg/clients/ecr/repository.go

```diff
@@ -125,3 +126,54 @@ func GenerateCreateRepositoryInput(name string, params *v1alpha1.RepositoryParam
 	}
 	return c
 }
+
+// CompareTags compares arrays of v1alpha1.Tag and ecr.Tag
+func CompareTags(tags []v1alpha1.Tag, ecrTags []ecr.Tag) bool {
+	if len(tags) != len(ecrTags) {
+		return false
+	}
+
+	SortTags(tags, ecrTags)
+
+	for i, t := range tags {
+		if t.Key != aws.StringValue(ecrTags[i].Key) || t.Value != aws.StringValue(ecrTags[i].Value) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// SortTags sorts array of v1alpha1.Tag and ecr.Tag on 'Key'
+func SortTags(tags []v1alpha1.Tag, ecrTags []ecr.Tag) {
+	sort.Slice(tags, func(i, j int) bool {
+		return tags[i].Key < tags[j].Key
+	})
+
+	sort.Slice(ecrTags, func(i, j int) bool {
+		return *ecrTags[i].Key < *ecrTags[j].Key
+	})
+}
+
+// DiffTags returns tags that should be added or removed.
+func DiffTags(spec []v1alpha1.Tag, current []ecr.Tag) (addTags []ecr.Tag, remove []string) {
+	addMap := map[string]string{}
+	for _, t := range spec {
+		addMap[t.Key] = t.Value
+	}
+	removeMap := map[string]string{}
+	for _, t := range current {
+		if addMap[aws.StringValue(t.Key)] == aws.StringValue(t.Value) {
+			delete(addMap, aws.StringValue(t.Key))
+			continue
+		}
+		removeMap[aws.StringValue(t.Key)] = aws.StringValue(t.Value)
+	}
+	for k, v := range addMap {
+		addTags = append(addTags, ecr.Tag{Key: aws.String(k), Value: aws.String(v)})
+	}
+	for k := range removeMap {
+		remove = append(remove, k)
+	}
```

Comment:
nitpick: can we pre-allocate the size of `addTags` and `remove` since they are known at this point?

Reply:
If we pre-allocate, we can't use `append` because it adds to the end so we'd need indexes in each of the for loops and that's not readily available because we iterate on map. So, I just preferred `append` instead of having to do bookkeeping separately.

Reply:
Just tested https://play.golang.org/p/jmLT-dbhnNA

Reply:
Right, I was suggesting:
```go
i := 0
for k := range removeMap {
  remove[i] = k
  i++
}
```

But I am fine with moving forward as-is if it is your preference.